src/core/chemstate_mod.F90

```diff
@@ -101,80 +107,191 @@ subroutine Chem_Allocate(Config, GridState, Species, ChemState, RC)
       RC = CC_SUCCESS
       ErrMsg = ''
       thisLoc = ' -> at Find_Number_of_Species (in core/chemstate_mod.F90)'
 
+
+      tmp = 0
+      do i = 1, ChemState%nSpecies
+         if (ChemState%ChemSpecies(i)%is_gas .eqv. .true.) then
+            tmp = tmp + 1
+         endif
+      enddo
+      ChemState%nSpeciesGas = tmp
+
+      tmp = 0
+      do i = 1, ChemState%nSpecies
+         if (ChemState%ChemSpecies(i)%is_aerosol .eqv. .true.) then
+            tmp = tmp + 1
+         endif
+      enddo
+      ChemState%nSpeciesAero = tmp
+
+      tmp = 0
+      do i = 1, ChemState%nSpecies
+         if (ChemState%ChemSpecies(i)%is_tracer .eqv. .true.) then
+            tmp = tmp + 1
+         endif
+      enddo
+      ChemState%nSpeciesTracer = tmp
+
+      tmp = 0
+      do i = 1, ChemState%nSpecies
+         if (ChemState%ChemSpecies(i)%is_dust .eqv. .true.) then
+            tmp = tmp + 1
+         endif
+      enddo
+      ChemState%nSpeciesDust = tmp
+
+      tmp = 0
+      do i = 1, ChemState%nSpecies
+         if (ChemState%ChemSpecies(i)%is_seasalt .eqv. .true.) then
+            tmp = tmp + 1
+         endif
+      enddo
+      ChemState%nSpeciesSeaSalt = tmp
       ! do nothing yet
       ! loop through Species etc and find the number of species
 
 
    end subroutine Find_Number_of_Species
 
-   !
-   !=========================================================================
-   !
-   subroutine Find_Index_of_Species(ChemState, Species, RC)
+   !> \brief Find the indices of species
+   !!
+   !! \param ChemState The ChemState object
+   !! \param RC The return code
+   !!
+   !! \ingroup core_modules
+   !!!>
+   subroutine Find_Indices_of_Species(ChemState, RC)
       ! USES
       USE Species_Mod,  ONLY : SpeciesType
 
       IMPLICIT NONE
 
       ! INOUT Params
       type(ChemStateType),  INTENT(INOUT) :: ChemState     ! chem State object
-      type(SpeciesType),   INTENT(INOUT) :: Species ! Species object
       ! OUTPUT Params
       INTEGER,             INTENT(OUT)   :: RC            ! Success or failure
 
       ! Error handling
       CHARACTER(LEN=255) :: ErrMsg
       CHARACTER(LEN=255) :: thisLoc
 
+      integer :: n ! looping variable
+
       ! Initialize
       RC = CC_SUCCESS
       ErrMsg = ''
-      thisLoc = ' -> at Find_Index_of_Species (in core/chemstate_mod.F90)'
-
-      ! do nothing yet
-      ! loop through Species etc and find the number of species
-
-
-   end subroutine Find_Index_of_Species
+      thisLoc = ' -> at Find_indices_of_Species (in core/chemstate_mod.F90)'
+
+      ! Find indices of all aerosol species
+      ALLOCATE(Chemstate%AeroIndex(ChemState%nSpeciesAero), STAT=RC)
+      IF ( RC /= CC_SUCCESS ) THEN
+         errMsg = 'Error allocating Chemstate%AeroIndex'
+         call CC_Error(errMsg, RC, thisLoc)
+         RETURN
+      ENDIF
+      do n = 1, ChemState%nSpecies
+         if (ChemState%ChemSpecies(n)%is_aerosol .eqv. .true.) then
+            Chemstate%AeroIndex(n) = n
```

Comment:
```suggestion
            Chemstate%AeroIndex(i) = n
```
I may be misinterpreting, but should this instead be something like this? Where `i` starts at 0 outside the loop and only increments within the `if`? (`ChemState%nSpeciesAero` should be less than (or equal to) `ChemState%nSpecies`, I would think.)